packages/loader/container-loader/src/contracts.ts

```diff
@@ -217,7 +217,7 @@ export const getPackageName = (
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	let containerPackageName: any;
 	if (codeDetails && "name" in codeDetails) {
-		containerPackageName = codeDetails;
+		containerPackageName = codeDetails.name;
```

Comment:
can we add a test to avoid similar regressions in the future?

Reply:
a unit test of just this function should be trivial to add.

Reply:
i'll add unit tests

Reply:
while adding the unit tests, I noticed the `codeDetails` types should include `IContainerPackageTypes`. Without this typing, the first `if` check would lose its meaning.

```ts
export const getPackageName = (
 codeDetails: IFluidCodeDetails | IContainerPackageInfo | undefined,
): IContainerPackageInfo => {
 // TODO: use a real type
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 let containerPackageName: any;
 if (codeDetails && "name" in codeDetails) {
  containerPackageName = codeDetails.name;
 } else if (isFluidPackage(codeDetails?.package)) {
  containerPackageName = codeDetails?.package.name;
 } else {
  containerPackageName = codeDetails?.package;
 }
 // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 return { name: containerPackageName };
};
```

`IContainerPackageName` is not passed today which probably means the first check should be removed. 

please correct me if i'm missing anything here.

Reply:
Removal sounds correct to me - if this has been "broken" for over 2 years, I think that's proof that it's unused and safe to remove.